### ISSUE TITLE
duplicate-key checker doesn't detect duplicate Enum keys

### ISSUE THREAD

**Issue by Neki** [PyCQA/pylint#5150]
### Bug description

The pylint 'duplicate-key' checker doesn't detect duplicate keys when said keys are enum members.

Consider the following code snippet:

```python
from enum import Enum

# shows that the duplicate-key checker is activated, but irrelevant for the actual issue
produces_a_warning = {"key": "value 1", "key": "value 2"}

class MyEnum(Enum):
    KEY = "key"

does_not_produce_a_warning = {MyEnum.KEY: "value 1", MyEnum.KEY: "value 2"}
```

Running pylint with the default rc file generates only one warning:

```
duplicate_keys.py:4:21: W0109: Duplicate key 'key' in dictionary (duplicate-key)
```

pylint should also have detected that the dictionary `does_not_produce_a_warning` includes a duplicated key.

### Command used

```shell
pylint duplicate_keys.py
```

### Pylint output

```shell
************* Module duplicate_keys
duplicate_keys.py:4:21: W0109: Duplicate key 'key' in dictionary (duplicate-key)
```

### Expected behavior

pylint should also have detected that the dictionary `does_not_produce_a_warning` includes a duplicated key, and should have produced two warnings.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.8.10 (default, Sep 28 2021, 16:10:42)
[GCC 9.3.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by Pierre-Sassoulas** [comment]
Thanks a lot for offering your help @jaydesl, I assigned you the issue :)
